documentation/design-docs/ipc-protocol.md

```diff
@@ -0,0 +1,183 @@
+# Event Pipe
+
+## Overview
+
+> TODO: Write an overview of what this does
+
+## TODOs
+* formalize struct names
+* formalize error codes (Use HRESULT values?)
+
+## Inter-Process Communication Specification
+
+### Current
+
+```c++
+enum class DiagnosticMessageType : uint32_t
+{
+    // EventPipe
+    StartEventPipeTracing = 1024, // To file
```

Comment:
If we are thinking about having the same protocol for a broad range of capabilities, maybe we should give these type a namespace?

Can we fix the naming of these types so that it does not need the comments?

Reply:
See my proposed idea of command_sets and commands below.  It sounds like what you are describing.